### ISSUE TITLE
[Bug]: Insert performance get worse than 2.0.2

### ISSUE THREAD

**Issue by yhmo** [milvus-io/milvus#17895]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: latest master branch with commit 5a85e6aaad230b2f3a6fa136c1af467ede22535d
- Deployment mode(standalone or cluster):
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Inset 20000 vectors with 1024 dim. The latest 2.1 totally cost 7.x seconds, the 2.0.2 cost 4.x seconds.

### Steps To Reproduce

```markdown
1. I compiled both 2.1 and 2.0.2 in local
2. use the following script to do insert:

import random
import time
import threading

from pymilvus import (
    connections,
    FieldSchema, CollectionSchema, DataType,
    Collection,
    utility
)


HOST = '127.0.0.1'
PORT = '19530'

# Const names
COLLECTION_NAME = 'demo4'
ID_FIELD_NAME = 'id_field'
VECTOR_FIELD_NAME = 'vector_field'

DIM = 1024
SHARD_NUM = 1

def create_collection():
    field1 = FieldSchema(name=ID_FIELD_NAME, dtype=DataType.INT64, is_primary=True, auto_id=False)
    field2 = FieldSchema(name=VECTOR_FIELD_NAME, dtype=DataType.FLOAT_VECTOR, dim=DIM)
    schema = CollectionSchema(fields=[field1, field2])
    collection = Collection(name=COLLECTION_NAME, data=None, schema=schema, shards_num=SHARD_NUM)
    print("\ncollection created:", COLLECTION_NAME)
    return collection

def prepare_data(count, id_start):
    data = [
        [i + id_start for i in range(count)],
        [[random.random() for _ in range(DIM)] for _ in range(count)],
    ]
    return data

def do_insert():
    collection = Collection(name=COLLECTION_NAME)
    BATCH_COUNT = 1000
    REPEAT = 20

    for t in range(REPEAT):
        data = prepare_data(BATCH_COUNT, t*BATCH_COUNT)
        time_start = time.time()
        collection.insert(data)
        time_end = time.time()
        time_cost = time_end - time_start
        print(" insert 1000 entities cost " + str(time_cost*1000) + " ms")


if __name__ == '__main__':
    connections.connect(host=HOST, port=PORT)

    collection = create_collection()

    time_start = time.time()
    do_insert()
    time_end = time.time()
    time_cost = time_end - time_start
    print("Totally cost " + str(time_cost) + " seconds")

    print("Total entities:", collection.num_entities)
```
```


### Milvus Log

_No response_

### Anything else?

_No response_

**Comment by yhmo** [comment]
The performance of v2.0.2:

```
collection created: demo4
 insert 1000 entities cost 136.11459732055664 ms
 insert 1000 entities cost 119.62532997131348 ms
 insert 1000 entities cost 110.75043678283691 ms
 insert 1000 entities cost 110.89181900024414 ms
 insert 1000 entities cost 112.35237121582031 ms
 insert 1000 entities cost 113.07811737060547 ms
 insert 1000 entities cost 112.07270622253418 ms
 insert 1000 entities cost 114.21728134155273 ms
 insert 1000 entities cost 113.6467456817627 ms
 insert 1000 entities cost 113.86680603027344 ms
 insert 1000 entities cost 117.73562431335449 ms
 insert 1000 entities cost 113.30342292785645 ms
 insert 1000 entities cost 111.28067970275879 ms
 insert 1000 entities cost 113.86561393737793 ms
 insert 1000 entities cost 112.5178337097168 ms
 insert 1000 entities cost 111.92107200622559 ms
 insert 1000 entities cost 122.32112884521484 ms
 insert 1000 entities cost 109.82847213745117 ms
 insert 1000 entities cost 111.08660697937012 ms
 insert 1000 entities cost 110.00776290893555 ms
Totally cost 4.219374179840088 seconds
Total entities: 20000

Process finished with exit code 0
```

**Comment by yhmo** [comment]
The performance of latest 2.1:

```
collection created: demo4
 insert 1000 entities cost 307.9855442047119 ms
 insert 1000 entities cost 244.47965621948242 ms
 insert 1000 entities cost 241.286039352417 ms
 insert 1000 entities cost 255.42020797729492 ms
 insert 1000 entities cost 240.2956485748291 ms
 insert 1000 entities cost 244.51732635498047 ms
 insert 1000 entities cost 247.7104663848877 ms
 insert 1000 entities cost 240.49687385559082 ms
 insert 1000 entities cost 244.7953224182129 ms
 insert 1000 entities cost 235.84556579589844 ms
 insert 1000 entities cost 244.97532844543457 ms
 insert 1000 entities cost 239.61472511291504 ms
 insert 1000 entities cost 247.33805656433105 ms
 insert 1000 entities cost 250.04315376281738 ms
 insert 1000 entities cost 744.7755336761475 ms
 insert 1000 entities cost 245.35393714904785 ms
 insert 1000 entities cost 239.26782608032227 ms
 insert 1000 entities cost 245.15771865844727 ms
 insert 1000 entities cost 235.0904941558838 ms
 insert 1000 entities cost 499.73273277282715 ms
Totally cost 7.7415876388549805 seconds
Total entities: 20000

Process finished with exit code 0
```

**Comment by xiaofan-luan** [comment]
/assign @longjiquan 
pls help on it

**Comment by longjiquan** [comment]
Strange, I also tried but the latency almost didn't change. Btw, I deployed Milvus using cluster mode.

2.1.0:
```
collection created: demo4
 insert 1000 entities cost 126.10816955566406 ms
 insert 1000 entities cost 104.71248626708984 ms
 insert 1000 entities cost 102.27108001708984 ms
 insert 1000 entities cost 105.30400276184082 ms
Totally cost 0.7601914405822754 seconds
Total entities: 4000
```

2.0.2:
```
collection created: demo4
 insert 1000 entities cost 126.73115730285645 ms
 insert 1000 entities cost 106.87756538391113 ms
 insert 1000 entities cost 99.1048812866211 ms
 insert 1000 entities cost 95.54314613342285 ms
Totally cost 0.7764227390289307 seconds
Total entities: 4000
```

**Comment by yhmo** [comment]
docker-compose.yml to start v2.0.2
```
version: '3.5'

services:
  etcd:
    container_name: milvus-etcd
    image: quay.io/coreos/etcd:v3.5.0
    environment:
      - ETCD_AUTO_COMPACTION_MODE=revision
      - ETCD_AUTO_COMPACTION_RETENTION=1000
      - ETCD_QUOTA_BACKEND_BYTES=4294967296
    volumes:
      - ${DOCKER_VOLUME_DIRECTORY:-.}/volumes/etcd:/etcd
    command: etcd -advertise-client-urls=http://127.0.0.1:2379 -listen-client-urls http://0.0.0.0:2379 --data-dir /etcd

  pulsar:
    container_name: milvus-pulsar
    image: apachepulsar/pulsar:2.8.2
    volumes:
      - ${DOCKER_VOLUME_DIRECTORY:-.}/volumes/pulsar:/pulsar/data
    environment:
      # bin/apply-config-from-env.py script will modify the configuration file based on the environment variables
      # nettyMaxFrameSizeBytes must be calculated from maxMessageSize + 10240 (padding)
      - nettyMaxFrameSizeBytes=104867840 # this is 104857600 + 10240 (padding)
      - defaultRetentionTimeInMinutes=10080
      - defaultRetentionSizeInMB=8192
      # maxMessageSize is missing from standalone.conf, must use PULSAR_PREFIX_ to get it configured
      - PULSAR_PREFIX_maxMessageSize=104857600
      - PULSAR_GC=-XX:+UseG1GC
    command: |
      /bin/bash -c \
      "bin/apply-config-from-env.py conf/standalone.conf && \
      exec bin/pulsar standalone --no-functions-worker --no-stream-storage"

  minio:
    container_name: milvus-minio
    image: minio/minio:RELEASE.2020-12-03T00-03-10Z
    environment:
      MINIO_ACCESS_KEY: minioadmin
      MINIO_SECRET_KEY: minioadmin
    volumes:
      - ${DOCKER_VOLUME_DIRECTORY:-.}/volumes/minio:/minio_data
    command: minio server /minio_data
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:9000/minio/health/live"]
      interval: 30s
      timeout: 20s
      retries: 3

  rootcoord:
    container_name: milvus-rootcoord
    image: milvusdb/milvus:v2.0.2
    command: ["milvus", "run", "rootcoord"]
    environment:
      ETCD_ENDPOINTS: etcd:2379
      MINIO_ADDRESS: minio:9000
      PULSAR_ADDRESS: pulsar://pulsar:6650
      ROOT_COORD_ADDRESS: rootcoord:53100
    depends_on:
      - "etcd"
      - "pulsar"
      - "minio"

  proxy:
    container_name: milvus-proxy
    image: milvusdb/milvus:v2.0.2
    command: ["milvus", "run", "proxy"]
    environment:
      ETCD_ENDPOINTS: etcd:2379
      MINIO_ADDRESS: minio:9000
      PULSAR_ADDRESS: pulsar://pulsar:6650
    ports:
      - "19530:19530"

  querycoord:
    container_name: milvus-querycoord
    image: milvusdb/milvus:v2.0.2
    command: ["milvus", "run", "querycoord"]
    environment:
      ETCD_ENDPOINTS: etcd:2379
      MINIO_ADDRESS: minio:9000
      PULSAR_ADDRESS: pulsar://pulsar:6650
      QUERY_COORD_ADDRESS: querycoord:19531
    depends_on:
      - "etcd"
      - "pulsar"
      - "minio"

  querynode:
    container_name: milvus-querynode
    image: milvusdb/milvus:v2.0.2
    command: ["milvus", "run", "querynode"]
    environment:
      ETCD_ENDPOINTS: etcd:2379
      MINIO_ADDRESS: minio:9000
      PULSAR_ADDRESS: pulsar://pulsar:6650
    depends_on:
      - "querycoord"

  indexcoord:
    container_name: milvus-indexcoord
    image: milvusdb/milvus:v2.0.2
    command: ["milvus", "run", "indexcoord"]
    environment:
      ETCD_ENDPOINTS: etcd:2379
      MINIO_ADDRESS: minio:9000
      PULSAR_ADDRESS: pulsar://pulsar:6650
      INDEX_COORD_ADDRESS: indexcoord:31000
    depends_on:
      - "etcd"
      - "pulsar"
      - "minio"

  indexnode:
    container_name: milvus-indexnode
    image: milvusdb/milvus:v2.0.2
    command: ["milvus", "run", "indexnode"]
    environment:
      ETCD_ENDPOINTS: etcd:2379
      MINIO_ADDRESS: minio:9000
      PULSAR_ADDRESS: pulsar://pulsar:6650
      INDEX_COORD_ADDRESS: indexcoord:31000
    depends_on:
      - "indexcoord"

  datacoord:
    container_name: milvus-datacoord
    image: milvusdb/milvus:v2.0.2
    command: ["milvus", "run", "datacoord"]
    environment:
      ETCD_ENDPOINTS: etcd:2379
      MINIO_ADDRESS: minio:9000
      PULSAR_ADDRESS: pulsar://pulsar:6650
      DATA_COORD_ADDRESS: datacoord:13333
    depends_on:
      - "etcd"
      - "pulsar"
      - "minio"

  datanode:
    container_name: milvus-datanode
    image: milvusdb/milvus:v2.0.2
    command: ["milvus", "run", "datanode"]
    environment:
      ETCD_ENDPOINTS: etcd:2379
      MINIO_ADDRESS: minio:9000
      PULSAR_ADDRESS: pulsar://pulsar:6650
    depends_on:
      - "datacoord"

networks:
  default:
    name: milvus
```

docker-compose.yml to start the latest dev build:
```
version: '3.5'

services:
  etcd:
    container_name: milvus-etcd
    image: quay.io/coreos/etcd:v3.5.0
    environment:
      - ETCD_AUTO_COMPACTION_MODE=revision
      - ETCD_AUTO_COMPACTION_RETENTION=1000
      - ETCD_QUOTA_BACKEND_BYTES=4294967296
    volumes:
      - ${DOCKER_VOLUME_DIRECTORY:-.}/volumes/etcd:/etcd
    command: etcd -advertise-client-urls=http://127.0.0.1:2379 -listen-client-urls http://0.0.0.0:2379 --data-dir /etcd

  pulsar:
    container_name: milvus-pulsar
    image: apachepulsar/pulsar:2.8.2
    volumes:
      - ${DOCKER_VOLUME_DIRECTORY:-.}/volumes/pulsar:/pulsar/data
    environment:
      # bin/apply-config-from-env.py script will modify the configuration file based on the environment variables
      # nettyMaxFrameSizeBytes must be calculated from maxMessageSize + 10240 (padding)
      - nettyMaxFrameSizeBytes=104867840 # this is 104857600 + 10240 (padding)
      - defaultRetentionTimeInMinutes=10080
      - defaultRetentionSizeInMB=8192
      # maxMessageSize is missing from standalone.conf, must use PULSAR_PREFIX_ to get it configured
      - PULSAR_PREFIX_maxMessageSize=104857600
      - PULSAR_GC=-XX:+UseG1GC
    command: |
      /bin/bash -c \
      "bin/apply-config-from-env.py conf/standalone.conf && \
      exec bin/pulsar standalone --no-functions-worker --no-stream-storage"

  minio:
    container_name: milvus-minio
    image: minio/minio:RELEASE.2020-12-03T00-03-10Z
    environment:
      MINIO_ACCESS_KEY: minioadmin
      MINIO_SECRET_KEY: minioadmin
    volumes:
      - ${DOCKER_VOLUME_DIRECTORY:-.}/volumes/minio:/minio_data
    command: minio server /minio_data
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:9000/minio/health/live"]
      interval: 30s
      timeout: 20s
      retries: 3

  rootcoord:
    container_name: milvus-rootcoord
    image: milvusdb/milvus-dev:2.1.0-20220704-f6ce0559
    command: ["milvus", "run", "rootcoord"]
    environment:
      ETCD_ENDPOINTS: etcd:2379
      MINIO_ADDRESS: minio:9000
      PULSAR_ADDRESS: pulsar://pulsar:6650
      ROOT_COORD_ADDRESS: rootcoord:53100
    depends_on:
      - "etcd"
      - "pulsar"
      - "minio"

  proxy:
    container_name: milvus-proxy
    image: milvusdb/milvus-dev:2.1.0-20220704-f6ce0559
    command: ["milvus", "run", "proxy"]
    environment:
      ETCD_ENDPOINTS: etcd:2379
      MINIO_ADDRESS: minio:9000
      PULSAR_ADDRESS: pulsar://pulsar:6650
    ports:
      - "19530:19530"

  querycoord:
    container_name: milvus-querycoord
    image: milvusdb/milvus-dev:2.1.0-20220704-f6ce0559
    command: ["milvus", "run", "querycoord"]
    environment:
      ETCD_ENDPOINTS: etcd:2379
      MINIO_ADDRESS: minio:9000
      PULSAR_ADDRESS: pulsar://pulsar:6650
      QUERY_COORD_ADDRESS: querycoord:19531
    depends_on:
      - "etcd"
      - "pulsar"
      - "minio"

  querynode:
    container_name: milvus-querynode
    image: milvusdb/milvus-dev:2.1.0-20220704-f6ce0559
    command: ["milvus", "run", "querynode"]
    environment:
      ETCD_ENDPOINTS: etcd:2379
      MINIO_ADDRESS: minio:9000
      PULSAR_ADDRESS: pulsar://pulsar:6650
    depends_on:
      - "querycoord"

  indexcoord:
    container_name: milvus-indexcoord
    image: milvusdb/milvus-dev:2.1.0-20220704-f6ce0559
    command: ["milvus", "run", "indexcoord"]
    environment:
      ETCD_ENDPOINTS: etcd:2379
      MINIO_ADDRESS: minio:9000
      PULSAR_ADDRESS: pulsar://pulsar:6650
      INDEX_COORD_ADDRESS: indexcoord:31000
    depends_on:
      - "etcd"
      - "pulsar"
      - "minio"

  indexnode:
    container_name: milvus-indexnode
    image: milvusdb/milvus-dev:2.1.0-20220704-f6ce0559
    command: ["milvus", "run", "indexnode"]
    environment:
      ETCD_ENDPOINTS: etcd:2379
      MINIO_ADDRESS: minio:9000
      PULSAR_ADDRESS: pulsar://pulsar:6650
      INDEX_COORD_ADDRESS: indexcoord:31000
    depends_on:
      - "indexcoord"

  datacoord:
    container_name: milvus-datacoord
    image: milvusdb/milvus-dev:2.1.0-20220704-f6ce0559
    command: ["milvus", "run", "datacoord"]
    environment:
      ETCD_ENDPOINTS: etcd:2379
      MINIO_ADDRESS: minio:9000
      PULSAR_ADDRESS: pulsar://pulsar:6650
      DATA_COORD_ADDRESS: datacoord:13333
    depends_on:
      - "etcd"
      - "pulsar"
      - "minio"

  datanode:
    container_name: milvus-datanode
    image: milvusdb/milvus-dev:2.1.0-20220704-f6ce0559
    command: ["milvus", "run", "datanode"]
    environment:
      ETCD_ENDPOINTS: etcd:2379
      MINIO_ADDRESS: minio:9000
      PULSAR_ADDRESS: pulsar://pulsar:6650
    depends_on:
      - "datacoord"

networks:
  default:
    name: milvus
```

**Comment by longjiquan** [comment]
![image](https://user-images.githubusercontent.com/31589260/177246449-df4b8fda-8088-4642-ac52-38d21858fc7f.png)

**Comment by longjiquan** [comment]
I added log for the entire insert path. I guess it's we upgrade the pulsar client that caused this issue.

**Comment by longjiquan** [comment]
![image](https://user-images.githubusercontent.com/31589260/177272974-914e8a31-5295-499f-ac52-d1cec323aa88.png)

**Comment by longjiquan** [comment]
Will be fixed by #18085 & #18090 .
/assign @yhmo 
/unassign

**Comment by yhmo** [comment]
I verify the issue on a Macbook with the latest dev build **milvusdb/milvus-dev:2.1.0-20220713-9059dba3**
Run the script on v2.0.2, the total time cost is 11 seconds.
Run the script on v2.1.0, the total time cost is 13 seconds.
Although there is a bit gap, I think it is acceptable. Close this issue.
